### PR TITLE
factor_list : check base and exp for coeff

### PR DESCRIPTION
**Pull request by Shekharrajak**
Fixes : https://github.com/sympy/sympy/issues/11198
### Current master branch

```
In [ ]: factor_list(sqrt(2)*sin(x), sin(x))
Out[ ]: (1, [(2, 1/2), (sin(x), 1)])

In [ ]: factor_list(3 ,x)
Out[ ]: (3, [])

In [ ]: factor_list(sqrt(3) ,x)
Out[ ]: (1, [(3, 1/2)])

```
### This branch

```
In [1]:  factor_list(sqrt(2)*sin(x), sin(x))
Out[1]: (√2, [(sin(x), 1)])

In [2]: factor_list(3 ,x)
Out[2]: (3, [])

In [5]:  factor_list(sqrt(3) ,x)
Out[5]: (√3, [])


```

### CODE REVIEW

**Comment by Shekharrajak** [comment]
Ping @asmeurer @jksuom

File: sympy/polys/polytools.py [hunk 1]
```diff
@@ -5652,6 +5652,9 @@ def _symbolic_factor_list(expr, opt, method):
             continue
         if arg.is_Pow:
             base, exp = arg.args
+            if base.is_Number and exp.is_Number:
```
Comment (smichr) [inline sympy/polys/polytools.py]: The more general thing to be doing (perhaps) is to use `if arg.is_number` to handle an expr like `sqrt(1+sqrt(2))`. (The base of that expression is a `number`, not a `Number`.

Comment (Shekharrajak) [inline sympy/polys/polytools.py]: Thanks @smichr . In line 5647 it should have 
```
if arg.is_number:
            coeff *= arg
```

Comment (Shekharrajak) [inline sympy/polys/polytools.py]: After this changes 
```
In [6]:  factor_list(sqrt(pi) ,x)
Out[6]: (√π, [])

In [7]:  factor_list(pi*sqrt(pi) ,x)
Out[7]: 
⎛ 3/2    ⎞
⎝π   , []⎠

```

Comment (Shekharrajak) [inline sympy/polys/polytools.py]: But it fails when we do
```
In [4]: g = y/(pi**2 + 2*pi + 1) + y*sin(x)/(pi**2 + 2*pi + 1)

In [5]: factor(g)
Out[5]: 
y⋅(sin(x) + 1)
──────────────
            2 
 1 + 2⋅π + π  
```
Ans: `y*(sin(x) + 1)/(pi + 1)**2`

**Comment by smichr** [comment]
good test: it tells us that factor should handle numeric expressions. So your change seems least invasive but perhaps could be relaxed with `or` instead of `and` for the Number checks. Perhaps you can think whether a power could be factored if either base or exponent were an expression and modify accordingly.

**Comment by Shekharrajak** [comment]
@smichr , Are you talking about this type of cases `factor_list((2**x) * sin(x))` ? Right now it returns same Error.

**Comment by Shekharrajak** [comment]
I think for now this PR is good to go. For more cases ( like for pi ) we need to factorize them similar to expression containing `symbol` ( to get (1+ pi)**2 for (1 + 2*pi + pi**2) coeff).

**Comment by smichr** [comment]
If this PR fails

```python
>>> g = y/(pi**2 + 2*pi + 1) + y*sin(x)/(pi**2 + 2*pi + 1)
>>> factor(g)
y*(sin(x) + 1)/(1 + pi)**2
```

then it should be fixed before committing. We shouldn't regress in one area while moving forward in another.

**Comment by Shekharrajak** [comment]
@smichr , that was the testcase which will fail when we use `is_number` , that testcase is [here](https://github.com/Shekharrajak/sympy/blob/793edddb634358fa27d0bb3244f94d8f07e73983/sympy/polys/tests/test_polytools.py#L2358) and passed in this PR.
When we use [`is_number`](https://github.com/sympy/sympy/pull/11201#discussion_r66085734) then [these cases](https://github.com/sympy/sympy/pull/11201#discussion_r66088740) works but not this 

```
>>> g = y/(pi**2 + 2*pi + 1) + y*sin(x)/(pi**2 + 2*pi + 1)
>>> factor(g)
y*(sin(x) + 1)/(1 + pi)**2
```

**Comment by smichr** [comment]
Thanks for clarifying...this is now in.

**Pull request merged**